.NET 4.6.1 project cannot reference a .NET Standard 1.4 project - conflicting types

Repro: https://github.com/gulbanana/repro-netstandard-httpclient
These two projects were created with VS2017 RC Refresh. Neither contains any meaningful code. `netstdlib` builds ok, but `netfxlib` fails:

```
banana@forsyth MINGW64 /c/code/repro-netstandard-httpclient/netfxlib (master)
$ dotnet build
Microsoft (R) Build Engine version 15.1.458.808
Copyright (C) Microsoft Corporation. All rights reserved.

  netstdlib -> C:\code\repro-netstandard-httpclient\netstdlib\bin\Debug\netstandard1.4\netstdlib.dll
C:\Program Files\dotnet\sdk\1.0.0-preview4-004233\Microsoft.Common.CurrentVersion.targets(1909,5): warning MSB3243: No way to resolve conflict between "System.IO.Compression, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" and "System.IO.Compression, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089". Choosing "System.IO.Compression, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" arbitrarily. [C:\code\repro-netstandard-httpclient\netfxlib\netfxlib.csproj]
C:\Program Files\dotnet\sdk\1.0.0-preview4-004233\Microsoft.Common.CurrentVersion.targets(1909,5): warning MSB3243: No way to resolve conflict between "System.Net.Http, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" and "System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a". Choosing "System.Net.Http, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" arbitrarily. [C:\code\repro-netstandard-httpclient\netfxlib\netfxlib.csproj]
  Consider app.config remapping of assembly "System.IO.Compression, Culture=neutral, PublicKeyToken=b77a5c561934e089" from Version "4.0.0.0" [C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6.1\System.IO.Compression.dll] to Version "4.1.0.0" [C:\Users\banana\.nuget\packages\system.io.compression\4.1.0\ref\net46\System.IO.Compression.dll] to solve conflict and get rid of warning.
  Consider app.config remapping of assembly "System.Net.Http, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" from Version "4.0.0.0" [C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6.1\System.Net.Http.dll] to Version "4.1.0.0" [C:\Users\banana\.nuget\packages\system.net.http\4.1.0\ref\net46\System.Net.Http.dll] to solve conflict and get rid of warning.
Class1.cs(9,29): error CS0433: The type 'HttpClient' exists in both 'System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' and 'System.Net.Http, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' [C:\code\repro-netstandard-httpclient\netfxlib\netfxlib.csproj]
```

Warnings are present even if you don't add a line of code to either project, and if you try to use any types from `System.IO.Compression` or `System.Net.Http` it's an unfixable compile error.

--- a/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets
+++ b/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets
@@ -304,6 +304,10 @@ Copyright (c) .NET Foundation. All rights reserved.
   <Target Name="ResolveLockFileReferences"
           DependsOnTargets="_ComputeLockFileReferences;_ComputeLockFileFrameworks">
     <ItemGroup>
+      <!-- Remove simple name references if we're directly providing a reference assembly to the compiler. For example,
+           consider a project with an Reference Include="System", and some NuGet package is providing System.dll -->
+      <Reference Remove="%(ResolvedCompileFileDefinitions.FileName)" />
+      
       <!-- Add the references we computed -->
       <Reference Include="@(ResolvedCompileFileDefinitions)" />
       <Reference Include="@(ResolvedFrameworkAssemblies)" />

--- a/test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildADesktopLibrary.cs
+++ b/test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildADesktopLibrary.cs
@@ -6,6 +6,7 @@ using Microsoft.NET.TestFramework.ProjectConstruction;
 using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Runtime.InteropServices;
 using System.Text;
 using System.Xml.Linq;
 using Xunit;
@@ -17,6 +18,11 @@ namespace Microsoft.NET.Build.Tests
         [Fact]
         public void It_can_use_HttpClient_and_exchange_the_type_with_a_NETStandard_library()
         {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return;
+            }
+
             var netStandardLibrary = new TestProject()
             {
                 Name = "NETStandardLibrary",

--- a/test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildADesktopLibrary.cs
+++ b/test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildADesktopLibrary.cs
@@ -1,0 +1,82 @@
+﻿using FluentAssertions;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Xml.Linq;
+using Xunit;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantToBuildADesktopLibrary : SdkTest
+    {
+        [Fact]
+        public void It_can_use_HttpClient_and_exchange_the_type_with_a_NETStandard_library()
+        {
+            var netStandardLibrary = new TestProject()
+            {
+                Name = "NETStandardLibrary",
+                TargetFrameworks = "netstandard1.4",
+                IsSdkProject = true
+            };
+
+            netStandardLibrary.SourceFiles["NETStandard.cs"] = @"
+using System.Net.Http;
+public class NETStandard
+{
+    public static HttpClient GetHttpClient()
+    {
+        return new HttpClient();
+    }
+}
+";
+
+            var netFrameworkLibrary = new TestProject()
+            {
+                Name = "NETFrameworkLibrary",
+                TargetFrameworks = "net461",
+                IsSdkProject = true
+            };
+
+            netFrameworkLibrary.ReferencedProjects.Add(netStandardLibrary);
+
+            netFrameworkLibrary.SourceFiles["NETFramework.cs"] = @"
+using System.Net.Http;
+public class NETFramework
+{
+    public void Method1()
+    {
+        System.Net.Http.HttpClient client = NETStandard.GetHttpClient();
+    }
+}
+";
+
+            var testAsset = _testAssetsManager.CreateTestProject(netFrameworkLibrary, "ExchangeHttpClient")
+                .WithProjectChanges((projectPath, project) =>
+                {
+                    if (Path.GetFileName(projectPath).Equals(netFrameworkLibrary.Name + ".csproj", StringComparison.OrdinalIgnoreCase))
+                    {
+                        var ns = project.Root.Name.Namespace;
+
+                        var itemGroup = new XElement(ns + "ItemGroup");
+                        project.Root.Add(itemGroup);
+
+                        itemGroup.Add(new XElement(ns + "Reference", new XAttribute("Include", "System.Net.Http")));
+                    }
+                })
+                .Restore(netFrameworkLibrary.Name);
+
+            var buildCommand = new BuildCommand(MSBuildTest.Stage0MSBuild, Path.Combine(testAsset.TestRoot, netFrameworkLibrary.Name));
+
+            buildCommand
+                .Execute()
+                .Should()
+                .Pass();
+
+        }
+    }
+}

